### ISSUE TITLE
Add adjoint method from the EventProp paper

### ISSUE THREAD

**Issue by Jegp** [norse/norse#159]
[This paper on EventProp](https://arxiv.org/abs/2009.08378) by @cpehle presents a super cool adjoint method for SNN optimisation.

We should definitely have this as the default gradient approach in Norse (as opposed to the current superspike approach).

**Comment by cpehle** [comment]
I agree (unsurprisingly) :). A good implementation of this also should actually also dramatically improve the memory footprint because we don't need to track the autograd graph that PyTorch builds and can only store data at spike events. We would also get rid of the matrix multiplication, which in the SuperSpike case propagates gradients of Neurons that are close to the threshold.
